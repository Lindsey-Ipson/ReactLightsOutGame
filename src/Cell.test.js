import React from "react";
import { render, screen } from "@testing-library/react";
import Cell from "./Cell";

describe("<Cell /> rendering", function () {
  let table, tbody, documentBody;

  beforeEach(function () {
    // Add a table and table row to the document so that TDs can go inside a TR
    table = document.createElement("table");
    tbody = table.createTBody();
    const row = tbody.insertRow();
    const cell = row.insertCell();
    documentBody = document.body.appendChild(table);
  });

  it("renders without crashing", function () {
    render(<Cell />, { container: documentBody.firstChild.firstChild });
  });

  it("matches snapshot when lit", function () {
    render(<Cell isLit={true} />, { container: documentBody.firstChild.firstChild });
    expect(documentBody).toMatchSnapshot();
  });

  it("matches snapshot when not lit", function () {
    render(<Cell isLit={false} />, { container: documentBody.firstChild.firstChild });
    expect(documentBody).toMatchSnapshot();
  });
});

