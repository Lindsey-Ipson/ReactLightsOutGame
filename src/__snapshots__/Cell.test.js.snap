// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> rendering matches snapshot when lit 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="Cell Cell-lit"
        role="button"
      />
    </tr>
  </tbody>
</table>
`;

exports[`<Cell /> rendering matches snapshot when not lit 1`] = `
<table>
  <tbody>
    <tr>
      <td
        class="Cell "
        role="button"
      />
    </tr>
  </tbody>
</table>
`;
